### PR TITLE
Abstracting data source

### PR DESCRIPTION
**Pull request by mshipton**
Data source defined with an interface. Sample static data source provided to test with.

### CODE REVIEW

File: model/src/main/java/com/emmaguy/hn/model/StaticDataSource.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * News data source based on static dummy data.
+ */
+public class StaticDataSource implements HackerNewsDataSource {
+
+    @Override
+    public void getLatestNewsItems() {
+        List<NewsItem> newsItems = new ArrayList<>();
+        for (int i=0; i<35; i++) {
```
Comment (emmaguy) [inline model/src/main/java/com/emmaguy/hn/model/StaticDataSource.java:16]: Can you format this file? Needs more spaces!

File: model/src/main/java/com/emmaguy/hn/model/StaticDataSource.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            newsItems.add(new NewsItem("Static news item #" + (i + 1), "http://www.google.com"));
```
Comment (emmaguy) [inline model/src/main/java/com/emmaguy/hn/model/StaticDataSource.java:17]: Also... Why not start i at 1? :smile:

File: model/src/main/java/com/emmaguy/hn/model/StaticDataSource.java (hunk 1 above)
Comment (mshipton) [inline model/src/main/java/com/emmaguy/hn/model/StaticDataSource.java:16]: Haha - of course!

File: model/src/main/java/com/emmaguy/hn/model/StaticDataSource.java (hunk 2 above)
Comment (mshipton) [inline model/src/main/java/com/emmaguy/hn/model/StaticDataSource.java:17]: Separating logic from presentation. Or something...

**Comment by emmaguy** [comment]
I'll handle the reformat - doing a refactor on another branch and I want this change too!

**Pull request merged**